doc/sql.extensions/README.blob_util.md

```diff
@@ -0,0 +1,184 @@
+# `RDB$BLOB_UTIL` package (FB 5.0)
+
+Before Firebird 5 BLOBs could be created appending data using the concatenate (`||`) operator with another BLOB.
+
+However there is two big problems with this approach:
+- BLOBs are always created in database page space, even if that is not required (not going to a persistent table).
+- Every time a concatenation happens, a new BLOB is created, filling the database with garbage.
+
+`RDB$BLOB_UTIL` package now exists to fix that problems.
+
+It allows BLOB creation in temporary page space (as well in the persistent page space) and has a workflow so that a BLOB is created and could have its data incrementally appended before it's closed and be available as a BLOB value. This is much like the BLOB client API works, but now available in PSQL.
+
+## Function `NEW`
+
+`RDB$BLOB_UTIL.NEW` is used to create a new BLOB. It returns a handle (an integer bound to the transaction) that should be used with the others functions of the package.
+
+Input parameter:
+ - `SEGMENTED` type `BOOLEAN NOT NULL`
+ - `TEMP_STORAGE` type `BOOLEAN NOT NULL`
+
+Return type: `INTEGER NOT NULL`.
+
+## Function `OPEN_BLOB`
+
+`RDB$BLOB_UTIL.OPEN_BLOB` is used to open an existing BLOB for read. It returns a handle that should be used with the others functions of the package.
+
+Input parameter:
+ - `BLOB` type `BLOB NOT NULL`
+
+Return type: `INTEGER NOT NULL`.
+
+## Procedure `APPEND`
+
+`RDB$BLOB_UTIL.APPEND` is used to append chunks of data to a BLOB handle created with `RDB$BLOB_UTIL.NEW`.
+
+Input parameters:
+ - `HANDLE` type `INTEGER NOT NULL`
+ - `DATA` type `VARBINARY(32767) NOT NULL`
+
```

Comment:
Perhaps we could create a VARIANT type which could be used as system routines arguments - and also as general data type.

System functions already can work in this way, but they do not have stores metadata.

Reply:
While the `VARIANT` type might be an interesting idea per se, it requires some serious thinking and discussions and it could be an overkill for this particular need if we need to release v5 really soon. So I'd be more happy with `APPEND_TEXT` (or `APPEND_STRING` if you wish) and separate `APPEND_BLOB`.